Quarkus : Error while trying to build the image: Unexpected internal error near index 1

I want to build a docker image with the plugin k8s from a quarkus project but it failed.

I create a project : https://github.com/survivant/quarkus-k8s-hello

I generate the project from codeio and added one endpoint to call kubernetes to retreive namespace.

I'm on Windows 10.

when I enter the command : 

mvn package k8s:build

```
C:\workspace\quarkus-k8s-hello>mvn package k8s:build -e
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------------< ca.demo:quarkus-k8s-hello >----------------------
[INFO] Building quarkus-k8s-hello 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ quarkus-k8s-hello ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ quarkus-k8s-hello ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ quarkus-k8s-hello ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\workspace\quarkus-k8s-hello\src\test\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ quarkus-k8s-hello ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ quarkus-k8s-hello ---
[INFO] Surefire report directory: C:\workspace\quarkus-k8s-hello\target\surefire-reports
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running ca.demo.HelloExampleTest
2020-03-19 10:29:13,457 INFO  [io.quarkus] (main) Quarkus 1.2.1.Final started in 1.410s. Listening on: http://127.0.0.1:8282
2020-03-19 10:29:13,460 INFO  [io.quarkus] (main) Profile test activated.
2020-03-19 10:29:13,460 INFO  [io.quarkus] (main) Installed features: [cdi, kubernetes-client, resteasy, resteasy-jsonb, smallrye-context-propagation, smallrye-fault-tolerance, smallrye-health, smallrye-metrics, smallrye-openapi, spring-boot-properties, swagger-ui]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.733 s - in ca.demo.HelloExampleTest
2020-03-19 10:29:14,807 INFO  [io.quarkus] (main) Quarkus stopped in 0.049s
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ quarkus-k8s-hello ---
[INFO]
[INFO] --- quarkus-maven-plugin:1.2.1.Final:build (default) @ quarkus-k8s-hello ---
[INFO] [org.jboss.threads] JBoss Threads version 3.0.0.Final
[INFO] Initializing dekorate session.
[INFO] Default s2i build generator....
[INFO] Registering s2i handler!
[INFO] Generating manifests.
[INFO] Processing kubernetes configuration.
[INFO] Processing openshift configuration.
[INFO] Processing s2i configuration.
[INFO] Closing dekorate session.
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building thin jar: C:\workspace\quarkus-k8s-hello\target\quarkus-k8s-hello-1.0.0-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 4446ms
[INFO]
[INFO] --- k8s-maven-plugin:0.2.0:build (default-cli) @ quarkus-k8s-hello ---
[INFO] k8s: Running in Kubernetes mode
[INFO] k8s: Building Docker image in Kubernetes mode
[INFO] k8s: Running generator quarkus
[ERROR] k8s: Failed to execute the build [Error while trying to build the image]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  19.607 s
[INFO] Finished at: 2020-03-19T10:29:24-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.jkube:k8s-maven-plugin:0.2.0:build (default-cli) on project quarkus-k8s-hello: Failed to execute the build: Error while trying to build the image: Unexpected internal error near index 1
[ERROR] \
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.jkube:k8s-maven-plugin:0.2.0:build (default-cli) on project quarkus-k8s-hello: Failed to execute the build
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to execute the build
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.buildAndTag (AbstractDockerMojo.java:718)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.processImageConfig (AbstractDockerMojo.java:526)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.executeBuildGoal (AbstractDockerMojo.java:685)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:79)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeDockerBuild (BuildMojo.java:107)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:52)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.jkube.kit.config.service.JKubeServiceException: Error while trying to build the image
    at org.eclipse.jkube.kit.config.service.kubernetes.DockerBuildService.build (DockerBuildService.java:54)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.buildAndTag (AbstractDockerMojo.java:715)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.processImageConfig (AbstractDockerMojo.java:526)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.executeBuildGoal (AbstractDockerMojo.java:685)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:79)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeDockerBuild (BuildMojo.java:107)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:52)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.util.regex.PatternSyntaxException: Unexpected internal error near index 1
\
    at java.util.regex.Pattern.error (Pattern.java:2027)
    at java.util.regex.Pattern.compile (Pattern.java:1788)
    at java.util.regex.Pattern.<init> (Pattern.java:1428)
    at java.util.regex.Pattern.compile (Pattern.java:1068)
    at java.lang.String.split (String.java:2317)
    at java.lang.String.split (String.java:2364)
    at org.eclipse.jkube.kit.build.core.assembly.JKubeTarArchiver.createArchive (JKubeTarArchiver.java:70)
    at org.eclipse.jkube.kit.build.core.assembly.DockerAssemblyManager.createBuildTarBall (DockerAssemblyManager.java:287)
    at org.eclipse.jkube.kit.build.core.assembly.DockerAssemblyManager.createDockerTarArchive (DockerAssemblyManager.java:178)
    at org.eclipse.jkube.kit.build.service.docker.ArchiveService.createArchive (ArchiveService.java:120)
    at org.eclipse.jkube.kit.build.service.docker.ArchiveService.createArchive (ArchiveService.java:115)
    at org.eclipse.jkube.kit.build.service.docker.BuildService.buildImage (BuildService.java:130)
    at org.eclipse.jkube.kit.build.service.docker.BuildService.buildImage (BuildService.java:77)
    at org.eclipse.jkube.kit.config.service.kubernetes.DockerBuildService.build (DockerBuildService.java:49)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.buildAndTag (AbstractDockerMojo.java:715)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.processImageConfig (AbstractDockerMojo.java:526)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.executeBuildGoal (AbstractDockerMojo.java:685)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:79)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.executeDockerBuild (BuildMojo.java:107)
    at org.eclipse.jkube.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:52)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

C:\workspace\quarkus-k8s-hello>

```

same thing

```
Microsoft Windows [version 10.0.18363.657]
(c) Microsoft Corporation, 2019. Tous droits réservés.

C:\workspace\quarkus-k8s-hello>
C:\workspace\quarkus-k8s-hello>mvn clean package -DskipTests=true k8s:build
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------------< ca.demo:quarkus-k8s-hello >----------------------
[INFO] Building quarkus-k8s-hello 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ quarkus-k8s-hello ---
[INFO] Deleting C:\workspace\quarkus-k8s-hello\target
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ quarkus-k8s-hello ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ quarkus-k8s-hello ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to C:\workspace\quarkus-k8s-hello\target\classes
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ quarkus-k8s-hello ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\workspace\quarkus-k8s-hello\src\test\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ quarkus-k8s-hello ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to C:\workspace\quarkus-k8s-hello\target\test-classes
[INFO]
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ quarkus-k8s-hello ---
[INFO] Tests are skipped.
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ quarkus-k8s-hello ---
[INFO] Building jar: C:\workspace\quarkus-k8s-hello\target\quarkus-k8s-hello-1.0.0-SNAPSHOT.jar
[INFO]
[INFO] --- quarkus-maven-plugin:1.3.0.Final:build (default) @ quarkus-k8s-hello ---
[INFO] [org.jboss.threads] JBoss Threads version 3.0.1.Final
[INFO] Initializing dekorate session.
[INFO] Default s2i build generator....
[INFO] Registering s2i handler!
[INFO] Registering s2i handler!
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building thin jar: C:\workspace\quarkus-k8s-hello\target\quarkus-k8s-hello-1.0.0-SNAPSHOT-runner.jar
[INFO] Generating manifests.
[INFO] Processing kubernetes configuration.
[INFO] Processing openshift configuration.
[INFO] Processing s2i configuration.
[INFO] Closing dekorate session.
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 3464ms
[INFO]
[INFO] --- k8s-maven-plugin:0.2.0:build (default-cli) @ quarkus-k8s-hello ---
[INFO] k8s: Running in Kubernetes mode
[INFO] k8s: Building Docker image in Kubernetes mode
[INFO] k8s: Running generator quarkus
[ERROR] k8s: Failed to execute the build [Error while trying to build the image]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  16.474 s
[INFO] Finished at: 2020-03-19T12:09:17-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.jkube:k8s-maven-plugin:0.2.0:build (default-cli) on project quarkus-k8s-hello: Failed to execute the build: Error while trying to build the image: Unexpected internal error near index 1
[ERROR] \
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

C:\workspace\quarkus-k8s-hello>
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,6 +25,7 @@ Usage:
 * Fix #53: Renamed plugins to openshift/kubernetes-maven-plugin keeping acronym (oc/k8s) for goal
 * Fix #97:  Port of fabric8io/fabric8-maven-plugin#1794 to fix ImageChange triggers not being set in DeploymentConfig when resource fragments are used
 * Ported PR fabric8io/fabric8-maven-plugin#1802, Labels are missing for some objects
+* Fix #112: Fix windows specific path error while splitting file path
 
 ### 0.2.0 (05-03-2020)
 * Fix #71: script to extract changelog information for notifications

--- a/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/core/assembly/JKubeTarArchiver.java
+++ b/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/core/assembly/JKubeTarArchiver.java
@@ -67,18 +67,11 @@ public class JKubeTarArchiver {
                 String targetFileName = entry.getValue();
 
                 // 1. Check whether nested directory is there, if not create it
-                String[] pathParts = targetFileName.split(File.separator);
-                File parentDirectory = inputDirectory;
-                if (pathParts.length > 0) {
-                    StringBuilder finalPathBuilder = new StringBuilder();
-                    for (int i = 0; i < pathParts.length - 1; i++) {
-                        String pathPart = pathParts[i];
-                        finalPathBuilder.append(pathPart + File.separator);
-                    }
-                    parentDirectory = new File(inputDirectory, finalPathBuilder.toString());
-                    parentDirectory.mkdirs();
+                File parentDirectory = new File(targetFileName).getParentFile();
+                if (parentDirectory != null) {
+                    FileUtil.createDirectory(new File(inputDirectory, parentDirectory.getPath()));
                 }
-                File targetFile = new File(parentDirectory, pathParts[pathParts.length - 1]);
+                File targetFile = new File(inputDirectory, targetFileName);
                 // Check whether file is not already created.
                 if (!targetFile.exists()) {
                     FileUtils.copyFile(srcFile, targetFile);

--- a/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/FileUtil.java
+++ b/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/FileUtil.java
@@ -32,11 +32,8 @@ import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.SimpleFileVisitor;
 import java.nio.file.attribute.BasicFileAttributes;
-import java.nio.file.attribute.PosixFilePermission;
 import java.util.ArrayList;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 import java.util.StringTokenizer;
 import java.util.stream.Stream;
 

--- a/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/FileUtilTest.java
+++ b/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/FileUtilTest.java
@@ -1,0 +1,92 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class FileUtilTest {
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Test
+    public void testCreateDirectory() throws IOException {
+        File newDirectory = new File(folder.getRoot(), "firstdirectory");
+        FileUtil.createDirectory(newDirectory);
+        assertTrue(newDirectory.exists());
+    }
+
+    @Test
+    public void testListFilesRecursively() throws IOException {
+        prepareDirectory();
+        List<File> fileList = FileUtil.listFilesRecursivelyInDirectory(folder.getRoot());
+        assertNotNull(fileList);
+        assertEquals(3, fileList.size());
+    }
+
+    @Test
+    public void testCleanDirectory() throws IOException {
+        prepareDirectory();
+        List<File> fileList = FileUtil.listFilesRecursivelyInDirectory(folder.getRoot());
+        assertEquals(3, fileList.size());
+        FileUtil.cleanDirectory(folder.getRoot());
+        assertFalse(folder.getRoot().exists());
+    }
+
+    @Test
+    public void trimWildCardCharactersFromPath() {
+        assertEquals("lib", FileUtil.trimWildcardCharactersFromPath("lib/**"));
+    }
+
+    @Test
+    public void testCopyDir() throws IOException {
+        prepareDirectory();
+        File copyTarget = new File(folder.getRoot(), "copyTarget");
+        FileUtil.copyDirectory(new File(folder.getRoot(), "foo"), copyTarget);
+
+        assertTrue(copyTarget.exists());
+        assertEquals(1, FileUtil.listFilesRecursivelyInDirectory(copyTarget).size());
+        assertTrue(new File(copyTarget, "fileInfoo1").exists());
+    }
+
+    @Test
+    public void testGetRelativePath() throws IOException {
+      prepareDirectory();
+      File relativeFile = FileUtil.getRelativePath(folder.getRoot(), new File(folder.getRoot(), "foo"));
+      assertEquals("foo", relativeFile.getPath());
+      relativeFile = FileUtil.getRelativePath(folder.getRoot(), new File(folder.getRoot().getAbsolutePath() + File.separator + "foo" + File.separator + "fileInfoo1"));
+      assertEquals("foo/fileInfoo1", relativeFile.getPath());
+    }
+
+    private void prepareDirectory() throws IOException {
+        File dir1 = folder.newFolder("foo");
+        File file1 = new File(dir1, "fileInfoo1");
+        folder.newFile("something");
+        File dir2 = folder.newFolder("bar");
+        File file2 = new File(dir2, "fileInfoo2");
+        file1.createNewFile();
+        file2.createNewFile();
+    }
+
+}

